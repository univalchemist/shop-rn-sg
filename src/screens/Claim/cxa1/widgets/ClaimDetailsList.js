import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import {
  Box,
  Text,
  PlainText,
  Divider,
  SectionHeadingText,
} from '@wrappers/components';
import theme from '@theme';

const Label = ({ text }) => {
  return (
    <Box
      py={1}
      px={2}
      mr={2}
      borderRadius={50}
      backgroundColor={theme.colors.gray[10]}
    >
      <PlainText fontSize={12} color={theme.colors.gray[8]}>
        {text}
      </PlainText>
    </Box>
  );
};

const ClaimDetailsList = ({ sectionsList }) => {
  const renderSection = (section, index) => (
    <Box key={`section-${index}`}>
      {section.title ? (
        <Box py={3}>
          <SectionHeadingText>{section.title}</SectionHeadingText>
        </Box>
      ) : null}

      {section.items.map((item, index) => (
        <Box key={`item-${index}`}>
          <Box py={3} key={`${item.label}`}>
            <Text>{item.label}</Text>
            <Text color={theme.colors.gray[0]}>{item.value || '-'}</Text>
            {item.tags?.length ? (
              <Box flexDirection="row" paddingTop={8}>
                {item.tags.map((tag, index) => (
                  <Label key={`tag-${index}`} text={tag} />
                ))}
              </Box>
            ) : null}
          </Box>
          <Divider full />
        </Box>
      ))}
    </Box>
  );

  return (
    <Box style={styles.container}>
      {sectionsList.map((section, index) => renderSection(section, index))}
    </Box>
  );
};

ClaimDetailsList.defaultProps = {
  sectionsList: [],
};

ClaimDetailsList.propTypes = {
  sectionsList: PropTypes.arrayOf(PropTypes.shape({})),
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
});

export default ClaimDetailsList;
