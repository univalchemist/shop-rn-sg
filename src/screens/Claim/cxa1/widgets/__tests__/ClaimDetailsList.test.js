import React from 'react';
import { renderForTest } from '@testUtils';

import ClaimDetailsList from '../ClaimDetailsList';

describe('ClaimDetailsList', () => {
  const sectionsList = [
    {
      title: 'Section title 1',
      items: [
        {
          label: 'Item 1 label',
          value: 'Item 1 value',
        },
        {
          label: 'Item 2 label',
          value: undefined,
        },
        {
          label: 'Item 2 label',
          value: undefined,
          tags: ['test'],
        },
      ],
    },
    {
      title: undefined,
      items: [
        {
          label: 'Item 3 label',
          value: 'Item 3 value',
        },
      ],
    },
  ];

  it('should render sections list', () => {
    const component = renderForTest(
      <ClaimDetailsList sectionsList={sectionsList} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
