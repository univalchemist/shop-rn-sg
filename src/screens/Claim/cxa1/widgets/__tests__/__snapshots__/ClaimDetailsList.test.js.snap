// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimDetailsList should render sections list 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "width": "100%",
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <View
                            py={3}
                            style={
                              Array [
                                Object {
                                  "paddingBottom": 16,
                                  "paddingTop": 16,
                                },
                              ]
                            }
                          >
                            <Text
                              color="gray.0"
                              fontFamily="default"
                              fontSize={16}
                              fontWeight="bold"
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#212121",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "fontWeight": "700",
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              Section title 1
                            </Text>
                          </View>
                          <View
                            style={
                              Array [
                                Object {},
                              ]
                            }
                          >
                            <View
                              py={3}
                              style={
                                Array [
                                  Object {
                                    "paddingBottom": 16,
                                    "paddingTop": 16,
                                  },
                                ]
                              }
                            >
                              <Text
                                color="gray.1"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#646464",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                Item 1 label
                              </Text>
                              <Text
                                color="#212121"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#212121",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                Item 1 value
                              </Text>
                            </View>
                            <View
                              backgroundColor="gray.5"
                              height={1}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "#e3e3e3",
                                    "height": 1,
                                    "width": "100%",
                                  },
                                ]
                              }
                              width="100%"
                            />
                          </View>
                          <View
                            style={
                              Array [
                                Object {},
                              ]
                            }
                          >
                            <View
                              py={3}
                              style={
                                Array [
                                  Object {
                                    "paddingBottom": 16,
                                    "paddingTop": 16,
                                  },
                                ]
                              }
                            >
                              <Text
                                color="gray.1"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#646464",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                Item 2 label
                              </Text>
                              <Text
                                color="#212121"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#212121",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                -
                              </Text>
                            </View>
                            <View
                              backgroundColor="gray.5"
                              height={1}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "#e3e3e3",
                                    "height": 1,
                                    "width": "100%",
                                  },
                                ]
                              }
                              width="100%"
                            />
                          </View>
                          <View
                            style={
                              Array [
                                Object {},
                              ]
                            }
                          >
                            <View
                              py={3}
                              style={
                                Array [
                                  Object {
                                    "paddingBottom": 16,
                                    "paddingTop": 16,
                                  },
                                ]
                              }
                            >
                              <Text
                                color="gray.1"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#646464",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                Item 2 label
                              </Text>
                              <Text
                                color="#212121"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#212121",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                -
                              </Text>
                              <View
                                flexDirection="row"
                                paddingTop={8}
                                style={
                                  Array [
                                    Object {
                                      "flexDirection": "row",
                                      "paddingTop": 8,
                                    },
                                  ]
                                }
                              >
                                <View
                                  backgroundColor="#ededed"
                                  borderRadius={50}
                                  mr={2}
                                  px={2}
                                  py={1}
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "#ededed",
                                        "borderRadius": 50,
                                        "marginRight": 8,
                                        "paddingBottom": 4,
                                        "paddingLeft": 8,
                                        "paddingRight": 8,
                                        "paddingTop": 4,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    color="#666666"
                                    fontFamily="default"
                                    fontSize={12}
                                    letterSpacing={0.3}
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#666666",
                                          "fontFamily": "Arial",
                                          "fontSize": 12,
                                          "letterSpacing": 0.3,
                                          "lineHeight": 22,
                                        },
                                      ]
                                    }
                                  >
                                    test
                                  </Text>
                                </View>
                              </View>
                            </View>
                            <View
                              backgroundColor="gray.5"
                              height={1}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "#e3e3e3",
                                    "height": 1,
                                    "width": "100%",
                                  },
                                ]
                              }
                              width="100%"
                            />
                          </View>
                        </View>
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {},
                              ]
                            }
                          >
                            <View
                              py={3}
                              style={
                                Array [
                                  Object {
                                    "paddingBottom": 16,
                                    "paddingTop": 16,
                                  },
                                ]
                              }
                            >
                              <Text
                                color="gray.1"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#646464",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                Item 3 label
                              </Text>
                              <Text
                                color="#212121"
                                fontFamily="default"
                                fontSize={16}
                                lineHeight={22}
                                style={
                                  Array [
                                    Object {
                                      "color": "#212121",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                Item 3 value
                              </Text>
                            </View>
                            <View
                              backgroundColor="gray.5"
                              height={1}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "#e3e3e3",
                                    "height": 1,
                                    "width": "100%",
                                  },
                                ]
                              }
                              width="100%"
                            />
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
